server/controllers/apply/applicationsController.ts

```diff
@@ -170,6 +170,28 @@ export default class ApplicationsController {
     }
   }
 
+  selectApplicationOrigin(): RequestHandler {
+    return async (req: Request, res: Response) => {
+      if ((req.body.applicationOrigin as ApplicationOrigin) === 'prisonBail') {
+        return res.redirect(paths.applications.searchByPrisonNumber({}))
+      }
+
+      if ((req.body.applicationOrigin as ApplicationOrigin) === 'courtBail') {
+        return res.redirect(paths.applications.searchByPrisonNumber({}))
+      }
+
+      const message = 'Please select an application type'
+
+      req.flash('errors', {
+        applicationOrigin: errorMessage('applicationOrigin', message),
+      })
+
+      req.flash('errorSummary', [buildErrorSummary('applicationOrigin', message)])
```

Comment:
do you think it's worth testing the error handling? (genuine q)

Reply:
I do actually, I think it's an easy regression to occur. I can address that in a separate PR if that's OK.